File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {

        if(s.length() != t.length()){
            return false;
        }

        Map<Character, Integer> storemap = new HashMap<>();

        for(int i = 0; i < s.length(); i++){
            storemap.put(s.charAt(i), storemap.getOrDefault(s.charAt(i), 0) + 1);
        }

        for(int i = 0; i < t.length(); i++){
            storemap.put(t.charAt(i), storemap.getOrDefault(t.charAt(i), 0) - 1);
        }

        for(int val: storemap.values()){
            if(val != 0){
                return false;
            }
        }
        return true;
    }
}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """

        #if sorted(s) == sorted(t):
        #    return True
        #return False

        if len(s) != len(t):
            return False

        anotherT = {}

        for char in s: 
            if char in anotherT:
                anotherT[char] += 1 
            else:
                anotherT[char] = 1
        
        for char in t:
            if char in anotherT:
                anotherT[char] -= 1 
            else:
                anotherT[char] = 1
        
        print(anotherT)
        for value in anotherT.values():
            if value != 0:
                return False
        return True
        #if len(set(anotherT.values())) == 1 and set(anotherT.values()).pop() == 0:
        #    return True
        #return False

       
        
        
        
        